pkg/docker/config/config_test.go

```diff
@@ -636,6 +637,42 @@ func TestGetAllCredentials(t *testing.T) {
 		require.Equal(t, d.password, conf.Password, "%v", d)
 	}
 
+	// test "credStore"
+	f, err := os.OpenFile(authFilePath, os.O_RDWR|os.O_TRUNC, 0644)
+	require.NoError(t, err)
+	_, err = f.Write([]byte(`{ "credsStore": "helper-registry" }`))
+	require.NoError(t, err)
+	err = f.Close()
+	require.NoError(t, err)
+	sys = types.SystemContext{
+		AuthFilePath:                authFilePath,
+		SystemRegistriesConfPath:    "",
+		SystemRegistriesConfDirPath: "",
+	}
+	authConfigs, err = GetAllCredentials(&sys)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(authConfigs))
+	require.Equal(t, "foo", authConfigs["registry-a.com"].Username)
+	require.Equal(t, "bar", authConfigs["registry-a.com"].Password)
+}
+
+func withTmpHome(t *testing.T) func() {
+	t.Helper()
+	dir, err := ioutil.TempDir("", "test-home")
+	if err != nil {
+		t.Fatal(err)
+		return func() {}
+	}
+	oldHome, oldHomeExists := os.LookupEnv("HOME")
+	os.Setenv("HOME", dir)
+	return func() {
+		if oldHomeExists {
+			os.Setenv("HOME", oldHome)
```

Comment:
I don’t immediately see that this is necessary (the existing `GetAllCredentials` test got by without), and anyway, AFAIK it doesn’t work; `containers/storage/pkg/homedir.Get()` caches the value throughout its lifetime, so a single-test override is either completely ineffective, or is effective only if that test happens to run first.

So, to make some of the code testable, we had to introduce `getCredentialsWithHomeDir` and I suspect something similar would be necessary in other cases that tried to override `HOME`.

Reply:
@mtrmac without this the test is failing on my machine because the  `GetAllCredentials` function also returns credentials from my docker config.

Reply:
I you don't have `$HOME/.docker/config.json` on your machine it probably works. But for me it's failing.

Reply:
>caches the value throughout its lifetime, so a single-test override is either completely ineffective, or is effective only if that test happens to run first.

I don't what to do about this.

Reply:
The test is touch not only testing config files but also actual user config files, that's not good.